### PR TITLE
Embed Google Map to show plan locations

### PR DESCRIPTION
**Pull request by rwangsc18**
## Description
Clearly describe changes in your PR, new feature or bug fixes

- Add an embedded Google Map object in plan show page
- Add markers based on trip places locations (lattitude, longitude) and place category (visit or eatory)

## Root Cause
Briefly describe the root cause and analysis of the problem

N/A

## Solution
Describe your code changes in detail for reviewers. Explain the solution and how it fixes the issue.
For new features, describe the high-level ideas.

- Add place locations (lat, lng) into the response of getPlanDetails request
- Follow Google Map API example and create a separate JS file "google_map.js" 
- Inside the JS file, initialize the map with markers for different places
- Note: some functions/codes are moved from "show-trip-details.js" into a separate script "show-trip-script.js"

## Result
image

## Testing
Did you add any new test for this change, performed manual integration tests, or both?

- Add Javascript tests based on Jest testing framework, see commit 76d4372
- Perform manual integration tests

## Final Checks
- [x] Have you removed commented code ?
- [x] Have you used gofmt and jsfmt to format your code ?
- [x] You have added unit tests

### CODE REVIEW

File: assets/js/show-trip-script.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Control location photos in slides
```
Comment (sananand007) [inline assets/js/show-trip-script.js:1]: can you add a small docstring at the top of the file like the one below, to describe what the File is doing? 
```
/**
 * [ description]
 */
```

File: assets/js/google_map.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  const latLngs = makeLatLngs(plan.LatLongs);
+  const map = new google.maps.Map(mapDiv, {
+    zoom: zoom,
+    center: findCenter(latLngs),
+  });
+  addMarkers(map, latLngs);
+}
+
+function makeLatLngs(arr) {
+  return arr.map(x => new LatLng(x[0], x[1]))
```
Comment (sananand007) [inline assets/js/google_map.js:23]: try .. catch here as well ?

**Review by sananand007 — APPROVED** [review]
lg

File: assets/js/show-trip-script.js (hunk 1 above)
Comment (rwangsc18) [reply assets/js/show-trip-script.js]: Sure, added in commit 82f56dd

File: assets/js/google_map.js (hunk 2 above)
Comment (rwangsc18) [reply assets/js/google_map.js]: updated in 82f56dd

File: assets/js/show-trip-script.js (hunk 1 above)
Comment (sananand007) [reply assets/js/show-trip-script.js]: Thanks :+1:

File: assets/js/google_map.js (hunk 2 above)
Comment (sananand007) [reply assets/js/google_map.js]: thanks :+1:

File: planner/planner_APIs.go [hunk 3]
```diff
@@ -95,6 +97,7 @@ type PlaceDetailsResp struct {
 	PhotoURL        string
 }
 
+// redundant?
 type TripShowDetails struct {
 	City    string `form:"city"`
 	Country string `form:"country"`
```
Comment (weihesdlegend) [inline planner/planner_APIs.go:103]: This seems to be redundant. Consider removing it.

File: assets/js/show-trip-details.js [hunk 4]
```diff
@@ -47,69 +38,44 @@ async function postPlanForUser() {
...
-    const url = new URL(pageURL)
-    const searchParams = new URLSearchParams(url.search);
-    searchParams.append('json_only', 'true')
-    url.search = searchParams.toString() 
-    return url.toString(); 
-}
+
 
 // Link to User Profile page
 profile.addEventListener("click", () => window.location = `/v1/users/${username}/profile`);
```
Comment (weihesdlegend) [inline assets/js/show-trip-details.js:81]: Add a EOF newline

File: assets/js/google_map.test.js [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+            { position: 1, label: "b" }
+        ];
+
+        // test
+        const result = makeOptions(latlngs, labels);
+        expect(result).toEqual(target);
+    })
+})
+
+
```
Comment (weihesdlegend) [inline assets/js/google_map.test.js:250]: Extra EOF newline.

File: assets/js/google_map.js [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+  addMarkers: addMarkers,
+  findCenter: findCenter,
+  LatLng: LatLng,
+  makeLatLngs: makeLatLngs,
+  makeMarkerLabels: makeMarkerLabels,
+  makeOptions: makeOptions,
+  Label: Label,
+  placeCategoryToIconCode: placeCategoryToIconCode,
+}
+module.exports = utils;
```
Comment (weihesdlegend) [inline assets/js/google_map.js:135]: Add a EOF newline

File: assets/js/google_map.test.js [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+    global.google = google; // mock
+});
+
+afterAll(() => {
+    global.google = origGoogle; // restore
+});
+
+describe("Test middle point", () => {
+    const { middlePoint } = utils;
+    test('Middle of an integer array [1,2,3]', () => {
```
Comment (weihesdlegend) [inline assets/js/google_map.test.js:17]: Should the function name be `arithmeticMean` or `mean`?

**Review by weihesdlegend — APPROVED** [review]
LGTM

File: planner/planner_APIs.go (hunk 3 above)
Comment (rwangsc18) [reply planner/planner_APIs.go]: Done

File: assets/js/show-trip-details.js (hunk 4 above)
Comment (rwangsc18) [reply assets/js/show-trip-details.js]: Done

File: assets/js/google_map.test.js (hunk 7 above)
Comment (rwangsc18) [reply assets/js/google_map.test.js:17]: Done

**Pull request merged**